src/shared/install.c

```diff
@@ -66,6 +69,31 @@ typedef struct {
         OrderedHashmap *have_processed;
 } InstallContext;
 
+typedef struct {
+        const char *pattern;
+        int action;
+} PresetRule;
+
+typedef struct {
+        PresetRule *rules;
+        size_t n_rules;
+} Presets;
+
+static inline void presets_freep(Presets *p) {
+        size_t i;
+
+        if (p) {
+                for (i = 0; i < p->n_rules; i++)
+                        if (p->rules[i].pattern)
+                                free((char *)p->rules[i].pattern);
+
+                free(p->rules);
+                p->n_rules = 0;
+        }
+}
+
+#define _cleanup_presets_ _cleanup_(presets_freep)
+
```

Comment:
yes, please just use _cleanup_(presets_freep), instead of defining your own shortcut macro that doesn't really cut anything short much...